features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/APIClientFactory.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+"use strict";
+import APIClient from "./APIClient";
+import Utils from "./Utils";
+
+class APIClientFactory {
+    constructor() {
+        if (APIClientFactory._instance) {
+            return APIClientFactory._instance;
+        }
+
+        this._APIClientMap = new Map();
+    }
+
+    getAPIClient() {
+        let environmentName = Utils.getEnvironment().label;
```

Comment:
Shall we move the getting environment name outside the `getAPIClient` method?
What we want to achieve through this factory method is, Get an environment specific swagger-client by giving the environment ID(label) and at the same time, preserve the swagger-client through multiple environments switches.

Reply:
i:e https://github.com/faif/python-patterns/blob/master/creational/factory_method.py

Reply:
Fixed with the commit https://github.com/RenukaFernando/carbon-apimgt/commit/f57e0a8b84ff8d8b27c3fe24f8c343348b770c8f